phoenix-hive-base/src/main/java/org/apache/phoenix/hive/PhoenixSerializer.java

```diff
@@ -167,6 +168,8 @@ public Writable serialize(Object values, ObjectInspector objInspector, DmlType d
                             value = ((HiveDecimal) value).bigDecimalValue();
                         } else if (value instanceof HiveChar) {
                             value = ((HiveChar) value).getValue().trim();
+                        } else if (CompatUtil.DateAndTimestampSupport()){
```

Comment:
Nice solution, but doesn't follow the java naming Conventions.
use .isDateAndTimeSupported() instead.